libbeat/tests/resources/goroutines.go

```diff
@@ -78,3 +74,32 @@ func CallAndCheckGoroutines(t testing.TB, f func()) {
 	f()
 	c.Check(t)
 }
+
+// WaitUntilOriginalCount waits until the original number of goroutines are
+// present before we has created the resource checker.
+func (c GoroutinesChecker) WaitUntilOriginalCount() int {
+	timeout := time.Now().Add(c.FinalizationTimeout)
+	var after int
+	for time.Now().Before(timeout) {
+		after = runtime.NumGoroutine()
+		if after <= c.before {
+			return 0
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	return after
+}
+
+func (c *GoroutinesChecker) RunFuncWhenNewGoroutinesAreStarted(f func()) {
+	for runtime.NumGoroutine() == c.before {
+		time.Sleep(10 * time.Millisecond)
+	}
```

Comment:
== `c.WaitUntilIncreased()`.

I think we can remove this. It's kinda of a duplicate to

```
c.WaitUntilIncreased()
... // go code block
```